pkg/cli/alpha/internal/update.go

```diff
@@ -0,0 +1,538 @@
+// It also validates if the version specified by the user is in a valid format and available for
+// download as a binary.
+func (opts *Update) Validate() error {
+	// Validate git repository
+	if err := opts.validateGitRepo(); err != nil {
+		return fmt.Errorf("failed to validate git repository: %w", err)
+	}
+
+	// Validate --from-branch
+	if err := opts.validateFromBranch(); err != nil {
+		return fmt.Errorf("failed to validate --from-branch: %w", err)
+	}
+
+	// Load the PROJECT configuration file
+	projectConfigFile, err := opts.loadConfigFile()
+	if err != nil {
+		return fmt.Errorf("failed to load the PROJECT file: %w", err)
+	}
+
+	// Extract the cliVersion field from the PROJECT file
+	opts.CliVersion = projectConfigFile.Config().GetCliVersion()
+
+	// Determine which Kubebuilder version to use for the update
+	if err := opts.defineFromVersion(); err != nil {
+		return fmt.Errorf("failed to define version: %w", err)
+	}
+
+	// Validate if the specified version is available as a binary in the releases
+	if err := opts.validateBinaryAvailability(); err != nil {
+		return fmt.Errorf("failed to validate binary availability: %w", err)
+	}
+
+	return nil
+}
+
+// Load the PROJECT configuration file to get the current CLI version
+func (opts *Update) loadConfigFile() (store.Store, error) {
+	projectConfigFile := yaml.New(machinery.Filesystem{FS: afero.NewOsFs()})
+	// TODO: assess if DefaultPath could be renamed to a more self-descriptive name
+	if err := projectConfigFile.LoadFrom(yaml.DefaultPath); err != nil {
+		if _, statErr := os.Stat(yaml.DefaultPath); os.IsNotExist(statErr) {
+			return projectConfigFile, fmt.Errorf("no PROJECT file found. Make sure you're in the project root directory")
+		}
+		return projectConfigFile, fmt.Errorf("fail to load the PROJECT file: %w", err)
+	}
+	return projectConfigFile, nil
+}
+
+// Define the version of the binary to be downloaded
+func (opts *Update) defineFromVersion() error {
+	// Allow override of the version from PROJECT file via command line flag
+	if opts.FromVersion != "" {
+		if !semver.IsValid(opts.FromVersion) {
+			return fmt.Errorf("invalid semantic version. Expect: vX.Y.Z (Ex: v4.5.0)")
+		}
+		opts.CliVersion = opts.FromVersion
+	}
+
+	if opts.CliVersion == "" {
+		return fmt.Errorf("failed to retrieve Kubebuilder version from PROJECT file. Please use --from-version to inform it")
+	}
+
+	return nil
+}
+
+// Validate if the version specified is available as a binary for download
+// from the releases
+func (opts *Update) validateBinaryAvailability() error {
+	// Ensure version has 'v' prefix for consistency with GitHub releases
+	if !strings.HasPrefix(opts.CliVersion, "v") {
+		opts.CliVersion = "v" + opts.CliVersion
+	}
+
+	// Construct the URL for pulling the binary from GitHub releases
+	opts.BinaryURL = fmt.Sprintf("https://github.com/kubernetes-sigs/kubebuilder/releases/download/%s/kubebuilder_%s_%s",
+		opts.CliVersion, runtime.GOOS, runtime.GOARCH)
```

Comment:
it need to be a const we will use it in more than a place ( to validate and to download the bin )